src/test/java/uk/gov/hmcts/reform/propertiesvolume/PropertiesvolumesSpringBootTest.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.reform.propertiesvolume;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import uk.gov.hmcts.reform.propertiesvolume.example.App;
+import uk.gov.hmcts.reform.propertiesvolume.example.HelloController;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = App.class,
+    properties = {
+        "spring.application.name=propertiesvolume-example",
+        "spring.cloud.propertiesvolume.enabled=true",
+        "spring.cloud.propertiesvolume.paths="
+            + PropertiesvolumesSpringBootTest.BASE_PATH + "/kvcreds-test"
+            + ","
+            + PropertiesvolumesSpringBootTest.BASE_PATH + "/kvcreds-test-other"
+            + ","
+            + PropertiesvolumesSpringBootTest.BASE_PATH + "/kvcreds-test-single/kv5",
+        }
+    )
+@AutoConfigureWebTestClient
+public class PropertiesvolumesSpringBootTest {
```

Comment:
maybe we should have a test that has some value substitutions in an application.yaml?

Reply:
this would prove the ordering from above :smile:

Reply:
Not sure I understand, do you mean the ordering of paths in 'spring.cloud.propertiesvolume.paths'?

Reply:
Could we have an application.yaml that uses the property substitution to check the ordering of the property sources ?
maybe something like this application.yaml
```yaml
someApplicationProperty:  "${secrets.draftstore.kvcreds-test}"
```